chimera_app/steam_config.py

```diff
@@ -0,0 +1,199 @@
+            data = vdf.VDFDict()
+            data['InstallConfigStore'] = {'Software': {'Valve': {'Steam': {}}}}
+
+        steam = data['InstallConfigStore']['Software']['Valve']['Steam']
+
+        if 'CompatToolMapping' not in steam:
+            steam['CompatToolMapping'] = {}
+        else:
+            stale_entries = []
+
+            for game in steam['CompatToolMapping']:
+                if ('name' in steam['CompatToolMapping'][game] and
+                        'config' in steam['CompatToolMapping'][game]):
+                    if (steam['CompatToolMapping'][game]['name'] == '' and
+                            steam['CompatToolMapping'][game]['config'] == ''):
+                        stale_entries.append(game)
+
+            for entry in stale_entries:
+                del steam['CompatToolMapping'][entry]
+
+        return data
+
+    @staticmethod
+    def load_local(path):
+        if file_exists(path):
+            data = vdf.load(open(path))
+        else:
+            data = vdf.VDFDict()
+            data['UserLocalConfigStore'] = {'Software': {'Valve': {'Steam': {}}}}
+
+        steam_input = data['UserLocalConfigStore']
+        if 'Apps' not in steam_input:
+            steam_input['Apps'] = {}
+
+        launch_options = data['UserLocalConfigStore']['Software']['Valve']['Steam']
+        if 'Apps' not in launch_options:
+            launch_options['Apps'] = {}
+
+        return data
+
+    @staticmethod
+    def download_tweaks_file(file_path, num_atempts=5, static_file=None):
+        """Downloads the latest tweak file from our server. It will try
+        num_atempts or fail. If the file can't be reached after num_atempts
+        then it will be replaced by a static file designated by static_file
+        unless the file already exists.
+
+        returns True if the procedure was successful by downloading the new
+        file or by replacing with static_file
+        """
+        url = ('https://raw.githubusercontent.com/ChimeraOS/'
+               'chimera/master/steam-tweaks.yaml')
+        ret_val = False
+        attempts = 0
+        while (attempts < num_atempts):
+            try:
+                r = requests.get(url, timeout=1)
+            except requests.Timeout:
+                continue
+            if r.status_code == 200:
+                open(file_path, 'wb').write(r.content)
+                ret_val = True
+                break  # We succeeded, no need to continue
+
+            time.sleep(1)
+            attempts += 1
+
+        if (not ret_val):
+            if (static_file is not None):
+                if file_exists(static_file):
+                    shutil.copyfile(static_file, file_path)
+                    ret_val = True
+                else:
+                    print(f'{file_path} does not exist')
+            else:
+                print(f'Could not get file from {url} in {num_atempts} atempts')
+        else:
+            print(f'Could not get file from {url} in {num_atempts} atempts')
```

Comment:
This case is triggered when `ret_val` is true. So wouldn't this mean the file was downloaded?

I think this logic could probably be simplified a bit.

Reply:
This else statement should go and don't print a message. The other cases are for when we want it to force the static file or not.